### PR TITLE
Add windows support for git-fastclone

### PR DESCRIPTION
**Pull request by DEMON1A**
Created a new function for the path generation that validates the current operation system before generating the path, and uses underscore in-case a windows system is being used to make sure git-fastclone doesn't use a non-allowed characters for windows filenames that prevents the submodules and lock files from being created

### CODE REVIEW

File: lib/git-fastclone.rb
Comment (justinseanmartin) [inline lib/git-fastclone.rb:56]: Two minor nits - 
1. I think repos will not infrequently have underscores in their names, so I'd suggest using something that makes the separator a bit more clear.
2. `s/portation_character/separator/`

```suggestion
      separator = if RbConfig::CONFIG['host_os'] =~ /mswin|msys|mingw|cygwin|bccwin|wince|emc/
                              '__'
                            else
                              ':'
                            end
      "#{separator}#{filename}"
```

File: lib/git-fastclone.rb
Comment (justinseanmartin) [inline lib/git-fastclone.rb:66]: The modern rubocop default is 120 for this. I think I'd rather we delete [this](https://github.com/square/git-fastclone/blob/master/.rubocop.yml#L19) config instead and let it pick up the default rather than adding a `rubocop:disable` for something relatively trivial like this.

**Review by justinseanmartin — COMMENTED** [review]
Two minor pieces of feedback, but happy to land after these changes, or happy to discuss if you have pushback.

File: lib/git-fastclone.rb
Comment (DEMON1A) [reply lib/git-fastclone.rb:56]: I can agree on both of these changes, I think it would make things more clear

File: lib/git-fastclone.rb
Comment (DEMON1A) [reply lib/git-fastclone.rb]: I removed the LineLength rule from robucop config, and removed the useless disable rule on the code, double check it should be resolved by now

**Review by justinseanmartin — APPROVED** [review]
Looks good, thanks for the contribution!

**Comment by justinseanmartin** [comment]
It looks like there are some minor rubocop violations to take care of. Once that's done, I'll merge this PR. Would you like me to cut a bugfix release with the changes?

**Comment by DEMON1A** [comment]
I can see where it fails, should combining the if statement as a one liner solve the issue? Something like this

```ruby
RbConfig::CONFIG['host_os'] =~ /mswin|msys|mingw|cygwin|bccwin|wince|emc/ ? '__' : ':'
```

Robucop didn't say anything about the code style when I was modifying git-fastclone, I think it's better if you correct those code style violations on your side if you don't mind, I can't really validate the code style locally

**Comment by justinseanmartin** [comment]
The rubocop fix looks like this:
```
diff --git a/lib/git-fastclone.rb b/lib/git-fastclone.rb
index 82326d5..18a67a7 100644
--- a/lib/git-fastclone.rb
+++ b/lib/git-fastclone.rb
@@ -49,10 +49,10 @@ module GitFastClone
 
     def reference_filename(filename)
       separator = if RbConfig::CONFIG['host_os'] =~ /mswin|msys|mingw|cygwin|bccwin|wince|emc/
-                              '__'
-                            else
-                              ':'
-                            end
+                    '__'
+                  else
+                    ':'
+                  end
       "#{separator}#{filename}"
     end
     module_function :reference_filename
```

It should produce this for you automatically if you run `bundle exec rubocop -a`. I don't have push access to your repo, but I could PR this to your repo if you can't apply the patch of get rubocop to run successfully locally.

**Comment by justinseanmartin** [comment]
Ok, git-fastclone 1.4.5 has been published with this fix.

**Pull request merged**
